apps/src/lib/util/GoogleAnalyticsReporter.js

```diff
@@ -0,0 +1,152 @@
+window.dataLayer = window.dataLayer || [];
+
+/**
+ * Global function used to track Google Universal events.
+ * It is a wrapper class that appends information to window.dataLayer
+ * This is the equivalent of ga() in Google Universal
+ */
+function gtag() {
+  window.dataLayer.push(arguments);
+}
+
+/**
+ * Class used to control the initialization of Google analytics
+ * for code.org properties.
+ * The class is responsible for
+ * - Adding the Google Tags needed by Tag manager
+ * - Send any dimensions information
+ * - Wrap Google Universal and Analytics functions for easier transition
+ * - Centralize common code
+ */
+class GoogleAnalyticsReporter {
+  enableGoogleAnalyticsUniversal = true;
+  enabledGoogleAnalytics4 = true;
```

Comment:
nit: inconsistent tense between `enable` and `enabled`.  Should be consistent.  I think maybe `isGoogleAnalytics4Enabled` makes for slightly more readable code where the value is consumed in other places.